File: TiaCodegen/Commands/Functions/NTrigCall.cs
using TiaCodegen.Commands.Coils;
using TiaCodegen.Commands.Signals;
using TiaCodegen.Interfaces;

namespace TiaCodegen.Commands.Functions
{
    public class NTrigCall : BaseNPCoil, IPartName
    {
        public NTrigCall(Signal signal) : base(signal, null, null)
        {
            PartName = "NBox";
        }

        public string PartName { get; set; }
    }
}

File: TiaCodegen-ts/src/Commands/Functions/NTrigCall.ts
import { Signal } from '../Signals/Signal.js';
import { IPartName } from '../../Interfaces/IPartName.js';
import { BaseNPCoil } from '../Coils/BaseNPCoil.js';

export interface NTrigCallOptions {
    signal: Signal;
}

export class NTrigCall extends BaseNPCoil implements IPartName {
    partName: string;

    constructor(options: NTrigCallOptions) {
        const { signal } = options;
        super(signal, null, null);
        this.partName = 'NBox';
    }
}
